### PR TITLE
add ASCII split function

### PR DESCRIPTION
**Pull request by tiankonglan**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?
no code

Issue Number: N/A

## What is the new behavior?

add ASCII split fuction
-
-

### CODE REVIEW

File: sources/ASCII.move [hunk 1]
```diff
@@ -107,4 +107,27 @@ module SFC::ASCII {
...
+            Vector::push_back(&mut buffer, byte);
+         } else {
+            Vector::push_back(&mut result, string(buffer));
+            buffer = Vector::empty<u8>();
+         };
+         i = i+1; 
+      };
+      if (Vector::length(&buffer) != 0) {
+         Vector::push_back(&mut result, string(buffer));
+      };
```
Comment (jiangying000) [inline sources/ASCII.move:130]: 调用 split(string(b"a878"), char(56))

返回的是：['a','7']
一般语言对这种case返回：['a','7','']

File: sources/ASCII.move [hunk 2]
```diff
@@ -107,4 +107,27 @@ module SFC::ASCII {
         byte >= 0x20 && // Disallow metacharacters
         byte <= 0x7E // Don't allow DEL metacharacter
     }
+
+    /// split string by char. Returns vector<String>
+   public fun split(string: String, char: Char) :  vector<String> {
```
[the remark this answers is not in the document]
Comment (jiangying000) [reply sources/ASCII.move:112]: 缩进少了个空格

File: integration-tests/ascii_test.move [hunk 3]
```diff
@@ -31,7 +31,59 @@ script {
...
+        if(Option::is_some<ASCII::String>(&string_op)){
+            let string = Option::destroy_some<ASCII::String>(string_op);
+            let result = ASCII::split(string,  ASCII::char(95u8)); // _ ansci is 95
+            assert!(Vector::length(&mut result) == 2, 120);
+        }; 
+    }
+}
+
+//# run --signers SFC
+//split string by char,  3th scence
```
Comment (jiangying000) [inline integration-tests/ascii_test.move:75]: scence -> scene ?

**Review by jiangying000 — CHANGES_REQUESTED** [review]

File: integration-tests/ascii_test.move [hunk 4]
```diff
@@ -31,7 +31,59 @@ script {
...
+    use SFC::ASCII;
+    use StarcoinFramework::Option;
+    use StarcoinFramework::Vector;
+    fun main(_sender: signer) {
+        // 1th scence
+        let bytes = b"a_b_c";
+        let string_op = ASCII::try_string(bytes);
+        if(Option::is_some<ASCII::String>(&string_op)){
+            let string = Option::destroy_some<ASCII::String>(string_op);
+            let result = ASCII::split(string,  ASCII::char(95u8)); // _ ansci is 95
```
Comment (jiangying000) [inline integration-tests/ascii_test.move:50]:
```suggestion
            let result = ASCII::split(string,  ASCII::char(95u8)); // _ ascii is 95
```

**Comment by WGB5445** [comment]
在代码完成后，可以用 `idea` 上的插件 格式化一下，统一下规范

**Review by WGB5445 — APPROVED** [review]

File: sources/ASCII.move [hunk 5]
```diff
@@ -107,4 +107,30 @@ module SFC::ASCII {
...
+        let i = 0;
+        let buffer = Vector::empty<u8>();
+        while ( i < len ) {
+            let byte = *Vector::borrow(&string.bytes, i);
+            if (byte != char.byte) {
+                Vector::push_back(&mut buffer, byte);
+            } else {
+                Vector::push_back(&mut result, string(buffer));
+                buffer = Vector::empty<u8>();
+                if (i == len - 1) {  // special
```
Comment (jiangying000) [inline sources/ASCII.move:124]: 如果 i == 0 呢

Comment (jiangying000) [reply sources/ASCII.move]: 哦，我搞错了

**Pull request merged**